neo/Consensus/ConsensusService.cs

```diff
@@ -252,10 +281,74 @@ private void ExtendTimerByFactor(int maxDelayInBlockTimes)
                 ChangeTimer(nextDelay);
         }
 
-        private void OnConsensusPayload(ConsensusPayload payload)
+        private void TryToSaveFuturePayloads(ConsensusPayload payload)
+        {
+            // Limit the maximum future payload to be saved, 
+            // avoiding caching payload when node is lagged behind by more than furtherPayloadToSave locks
+            uint furtherPayloadToSave = 10;
+            if ((payload.BlockIndex >= context.Block.Index + 1) && (payload.BlockIndex <= context.Block.Index + furtherPayloadToSave))
+            {
+                ConsensusMessage futureMessage;
+                try
+                {
+                    futureMessage = payload.ConsensusMessage;
+                }
+                catch (FormatException)
+                {
+                    return;
+                }
+                catch (IOException)
+                {
+                    return;
```

Comment:
How can these exceptions happen? Shouldn't we log them?

Reply:
This is not related to this PR, it is original from the current code. Maybe better to open an issue to discuss that.